src/worker/trainer.py

```diff
@@ -51,8 +51,9 @@ def _run_and_optimize_model(self, data):
 
         elif self.optimize_strategy == 'GAN':
             total_loss = 0
-            for network_name in self.model.network_names:
-                self.optimizers[network_name].zero_grad()
+            for optimizer_name in self.optimizers.keys():
+                self.optimizers[optimizer_name].zero_grad()
+                network_name = global_config['optimizers'][optimizer_name]['target_network']
```

Comment:
Does validator need the same change?

Reply:
Yes, the `network_name` can not be determined using `optimizer_name` during validation.
This issue should be solved together with the validation metrics of GAN